src/core/border_router/nd6_types.hpp

```diff
@@ -58,7 +58,7 @@ using ot::Encoding::BigEndian::HostSwap32;
 
 namespace ot {
 namespace BorderRouter {
-namespace RouterAdv {
+namespace Nd6 {
```

Comment:
Should these definitions be moved to https://github.com/openthread/openthread/blob/main/src/core/net/icmp6.hpp ?

Or at least under a `Ip6::Icmp6` namespace?

Reply:
I moved it under `Ip6::Nd` namespace (to be similar to `Ip6::Udp`, `Ip6::Tcp` and `Ip6::Icmp`, etc.).
Moved the src files also under `net` folder.

Reply:
Do we want to move `Ip6::Nd` under `Ip6::Icmp`? Given that ND is built on top of ICMPv6?

Reply:
Yeah, I'm fine with how it is for now. If we expand ND further, we may want to consider aligning more with the protocol hierarchy.